### ISSUE TITLE
Improvement: Default values & Undefined tool

### ISSUE THREAD

**Issue by LMS0815** [kantlivelong/OctoPrint-SmartPreheat#6]
My gcode files do not provide the tool ID. Please consider searching for tool set gcode without `T` parameter

```
M104 Sxx
M109 Sxx
```

I recommend adding tools ID **-1** to cover undefined (= last use) tool to work with the snippet below.
This code is in addition providing default values, if the according variables are not set.

```
{# Init vars #}
{%- if plugins.smartpreheat is defined %}
 {%- set bed, list = plugins.smartpreheat.bed|default(75 , true), plugins.smartpreheat.tools|default({-1: 195} , true) %}
{%- endif -%}

; Set bed
{%- if printer_profile.heatedBed %} 
M190 S{{- (bed * 0.8)|round|int -}} ; Wait for Bed
M140 S{{- bed -}} ; Set Bed
{%- endif %}

; Set tool temps
{%- for tool, temp in list.items() %}
M104 {{- '' if tool < 0 else ' T' + tool|string }} S{{- temp -}} ; Set Hotend
{%- endfor %}

; Wait bed
{%- if printer_profile.heatedBed %} 
M190 S{{- bed -}} ; Wait for Bed
{%- endif %}

; Wait tool temps
{%- for tool, temp in list.items() %}
M109 {{- '' if tool < 0 else ' T' + tool|string }} S{{- temp -}} ; Wait for Hotend
{%- endfor %}
```

Feel free to use it. Might be nice to be mentioned as the source :)

**Comment by kantlivelong** [comment]
Apparently this repo got marked as unwatched for me. Doh!

A few bugs with the parser were just fixed in 0.0.3 which may have resulted in temps set later in the gcode file to be used.

Can you upgrade and try again. Please provide logs as detailed in the ticket submission process.

**Comment by LMS0815** [comment]
Kindly always provide valid default values (=temperatures), even if no scanning occurred.
